app/main.py

```diff
@@ -1,3 +1,17 @@
 def cache(func):
-    # Write your code here
-    pass
+    saved = {}
+
+    def inner(*args):
+        func_with_args = (func, args)
```

Comment:
You don't have to use a func object, dict `saved` is separate for each decorated function